src/main/java/bot/listeners/moderation/SpamControl.java

```diff
@@ -34,49 +32,50 @@ public SpamControl(Config config) {
-                    MessageEmbed embed = new EmbedBuilder().setDescription("You have been muted for **5 minutes** for spamming in a channel! You have also been given **1 warning**!").setFooter(member.getEffectiveName()).build();
-                    c.sendMessageEmbeds(embed).queue(m -> {
-                            },
-                            e -> event.getChannel().sendMessageEmbeds(embed).queue()
-                    );
-                });
-                event.getChannel().sendMessage("Please do not spam! You have been muted for `5 minutes` and given `1 warning`!").queue();
-                if (Tools.memberToWarns.computeIfAbsent(member, (m) -> new ArrayList<>()).size() >= 4) {
-                    event.getChannel().sendMessage("Kicked " + member.getAsMention() + " from the server because they exceeded `3 warnings`!").queue();
-                    member.kick("Exceeded 3 warnings!").queue();
-                }
-                Tools.muteMember(member, event.getGuild(), 5, "Spamming");
-            } else if (System.currentTimeMillis() - lastTimeSent > scseconds * 1000) {
-                messageTracking.get(member).msgNum = 1;
-                messageTracking.get(member).lastTimeSent = System.currentTimeMillis();
+        MessageHistory history = messageTracking.get(id);
+        long time = System.currentTimeMillis();
+        if (history == null) {
+            messageTracking.put(id, new MessageHistory(1, time));
+            return;
+        }
+        history.msgNum++;
+        if (history.msgNum == scmessages && time - history.lastTimeSent <= scseconds * 1000) {
+            MessageEmbed embed = new EmbedBuilder()
+                .setDescription("You have been muted for **5 minutes** for spamming in a channel! You have also been given **1 warning**!")
+                .setFooter(member.getEffectiveName())
+                .build();
+            member.getUser()
+                .openPrivateChannel()
+                .flatMap(c -> c.sendMessageEmbeds(embed))
+                .onErrorFlatMap(e -> event.getChannel().sendMessageEmbeds(embed))
+                .queue();
+            event.getChannel().sendMessage("Please do not spam, " + event.getAuthor().getAsMention() + "! You have been muted for `5 minutes` and given `1 warning`!").queue();
+            if (Tools.getWarns(member.getId()).size() >= 4) {
+                event.getChannel().sendMessage("Kicked " + member.getAsMention() + " from the server because they exceeded `3 warnings`!").queue();
+                member.kick("Exceeded 3 warnings!").queue();
             }
-            messageTracking.get(member).lastTimeSent = System.currentTimeMillis();
+            Tools.muteMember(member, 5, "Spamming");
+        } else if (time - history.lastTimeSent > scseconds * 1000) {
+            history.msgNum = 1;
+            history.lastTimeSent = time;
         }
     }
 
+    // TODO: is this necessary? We have the constructor to add blacklists
```

Comment:
The event in this is necessary cause We don't want to manually add every id, But we don't need it for now.